python/GafferUI/WidgetAlgo.py

```diff
@@ -65,14 +65,19 @@ def joinEdges( widgets, orientation = None ) :
 	l = len( visibleWidgets )
 	for i, widget in enumerate( visibleWidgets ) :
 
-		if isinstance( widget, GafferUI.BoolPlugValueWidget ) :
-			# Special case - we need to apply the rounding to
-			# the internal widget.
-			## \todo Is there a better approach here, perhaps
-			# using the stylesheet?
-			qtWidget = widget.boolWidget()._qtWidget()
-		else :
-			qtWidget = widget._qtWidget()
+		qtWidget = widget._qtWidget()
+
+		# Special case - many Gaffer Widgets place a single Widget inside a QGridLayout
+		# with a single item - in this case we need to apply the rounding to
+		# the internal widget.  Searching for the internal widget feels a little hacky,
+		# but it's more general than only supporting BoolPlugValueWidget, which is what
+		# we were doing before
+		## \todo Is there a better approach here, perhaps
+		# using the stylesheet?
+		if type( qtWidget ) == QtWidgets.QWidget and qtWidget.layout():
+			layoutItem = qtWidget.layout().itemAt( 0 )
+			if layoutItem  and layoutItem.widget():
+				qtWidget = layoutItem.widget()
```

Comment:
It seemed like you might not have been completely happy with this heuristic, so I had a stab at my own : https://github.com/johnhaddon/gaffer/commit/5b37fdf43514ab516b376d4cf110b7481e1862ba. I think maybe that feels slightly more general?